satpy/readers/viirs_edr_active_fires.py

```diff
@@ -43,11 +49,22 @@ def get_dataset(self, dsid, dsinfo):
             Dask DataArray: Data
 
         """
-        data = self[dsinfo.get('file_key', dsid.name)]
+
+        key = dsinfo.get('file_key', dsid.name).format(variable_prefix=self.prefix)
+        data = self[key]
         data.attrs.update(dsinfo)
 
-        data.attrs["platform_name"] = self['/attr/satellite_name']
-        data.attrs["sensor"] = self['/attr/instrument_name']
+        if self.filename_info['satellite_name'].upper() == "NPP":
```

Comment:
Use a dictionary instead please and no print statements (that's what KeyError exceptions are for).